stix2/pattern_expressions.py

```diff
@@ -0,0 +1,223 @@
+            super(IsSubsetComparisonExpression, self).__init__("ISSUBSET", lhs, rhs, negated)
+
+
+class IsSupersetComparisonExpression(ComparisonExpression):
+        def __init__(self, lhs, rhs, negated=False):
+            super(IsSupersetComparisonExpression, self).__init__("ISSUPERSET", lhs, rhs, negated)
+
+
+class BooleanExpression(PatternExpression):
+    def __init__(self, operator, operands):
+        self.operator = operator
+        self.operands = []
+        for arg in operands:
+            if not hasattr(self, "root_type"):
+                self.root_type = arg.root_type
+            elif self.root_type and (self.root_type != arg.root_type) and operator == "AND":
+                raise ValueError("This expression cannot have a mixed root type")
+            elif self.root_type and (self.root_type != arg.root_type):
+                self.root_type = None
+            self.operands.append(arg)
+
+    def __str__(self):
+        sub_exprs = []
+        for o in self.operands:
+            sub_exprs.append("%s" % o)
+        return (" " + self.operator + " ").join(sub_exprs)
+
+
+class AndBooleanExpression(BooleanExpression):
+    def __init__(self, operands):
+        super(AndBooleanExpression, self).__init__("AND", operands)
+
+
+class OrBooleanExpression(BooleanExpression):
+    def __init__(self, operands):
+        super(OrBooleanExpression, self).__init__("OR", operands)
+
+
+class ObservableExpression(PatternExpression):
+    def __init__(self, operand):
+        self.operand = operand
+
+    def __str__(self):
+        return "[%s]" % self.operand
+
+
+class CompoundObservableExpression(PatternExpression):
+    def __init__(self, operator, operands):
+        self.operator = operator
+        self.operands = operands
+
+    def __str__(self):
+        sub_exprs = []
+        for o in self.operands:
+            sub_exprs.append("%s" % o)
+        return (" " + self.operator + " ").join(sub_exprs)
+
+
+class AndObservableExpression(CompoundObservableExpression):
+    def __init__(self, operands):
+        super(AndObservableExpression, self).__init__("AND", operands)
+
+
+class OrObservableExpression(CompoundObservableExpression):
+    def __init__(self, operands):
+        super(OrObservableExpression, self).__init__("OR", operands)
+
+
+class FollowedByObservableExpression(CompoundObservableExpression):
+    def __init__(self, operands):
+        super(FollowedByObservableExpression, self).__init__("FOLLOWEDBY", operands)
+
+
+class ParentheticalExpression(PatternExpression):
+    def __init__(self, exp):
+        self.expression = exp
+        if hasattr(exp, "root_type"):
+            self.root_type = exp.root_type
+
+    def __str__(self):
+        return "(%s)" % self.expression
+
+
+class ExpressionQualifier(PatternExpression):
+    pass
+
+
+class RepeatQualifier(ExpressionQualifier):
+    def __init__(self, times_to_repeat):
+        if isinstance(times_to_repeat, IntegerConstant):
```

Comment:
Should not use IntegerConstant here (or in `WithinQualifier` or `StartStopQualifier`. These are not constants in the same way that the RHS of a comparison expression is. Just an `int` should be fine.